rewrite-core/src/test/java/org/openrewrite/config/YamlResourceLoaderTest.java

```diff
@@ -275,4 +275,81 @@ void declarativeAttributionIncludesRecipeListContributors() {
         Optional<Contributor> maybeJon = recipe.getContributors().stream().filter(c -> c.getName().equals("Jonathan Schneider")).findFirst();
         assertThat(maybeJon).isPresent();
     }
+
+    @Test
+    void recipeExamples() {
+        Environment env = Environment.builder()
+          .load(new YamlResourceLoader(new ByteArrayInputStream(
+            //language=yml
+            """
+              type: specs.openrewrite.org/v1beta/recipe
+              name: test.ChangeTextToHello
+              displayName: Change text to hello
+              recipeList:
+                  - org.openrewrite.text.ChangeText:
+                      toText: Hello!
+              """.getBytes()
+          ), URI.create("rewrite.yml"), new Properties()))
+          .load(new YamlResourceLoader(new ByteArrayInputStream(
+            //language=yml
+            """
+              type: specs.openrewrite.org/v1beta/example
+              recipeName: test.ChangeTextToHello
+              examples:
+                - description: "Change World to Hello in a text file"
+                  before: "World"
+                  after: "Hello!"
+                  language: "text"
+                - description: "Change World to Hello in a java file"
+                  before: |
+                    public class A {
+                        void method() {
+                            System.out.println("world");
```

Comment:
Nitpick: case sensitive?
```suggestion
                            System.out.println("World");
```